docs/spatial-index-guide.md

```diff
@@ -0,0 +1,94 @@
+
+The grid and point relationship is like that black point is the middle of a grid, the red dot is just inside the grid. The red point is inside the grid, it can be replaced by the center point of the grid, indicating that the point is within the grid. Therefore, the coordinates of points in a grid are replaced by black points in the middle. This is the characteristic of data load.  At the same time of data load, carbondata will generate hash ID according to the coordinates of rows and columns of the grid. These hash IDs are the same as Z order when querying. Detailed conversion algorithm can refer to the design documents of spatial index.
+
+![File Directory Structure](../docs/images/spatial-index-1.png?raw=true)
+
+When querying, the user enters the true space polygon coordinates, carbondata use the polygon and spatial region information passed in when creating a table build a quad tree. The nodes in the quad tree are composed of hash ids generated by the row and column information projected in the polygon area and group photo in map area. When the query polygon area is not disjon from the grid center point, the grid is considered selected.  In the following figure, user select a quadrilateral polygon,  The grid with the center point in the region will generate a quadtree. A list of line with continuous properties will be generated in the query process, like [97->97  99->99  102->102  104->111  120->120  122->123  151->151  157->158  159->159  192->208  210->210  216->216  225->225  228->229], each part of the list represents a continuous grid area. Carbondata use that line list to prune and filtered. About the detail can be search under https://issues.apache.org/jira/browse/CARBONDATA-3548
+
+![File Directory Structure](../docs/images/spatial-index-2.png?raw=true)
+
+
+
+## Installation and Deployment
+
+Build source with modules geo open, can open "pom.xml" and check whether the mode has been open. 
+
+![File Directory Structure](../docs/images/spatial-index-3.png?raw=true)
+
+Then you can get the "carbondata-geo-2.0.0-SNAPSHOT.jar" keep this jar and 'jst-core.jar' to your carbonlib path.
+
+## Basic Command
+
+### Create Table
+
+spatial index need to appoint the source column and other regional information. carbon will create a Invisible hash id column.
+
+example
+
+```
+create table source_index(id BIGINT, latitude long, longitude long) stored by 'carbondata' TBLPROPERTIES (
+'INDEX_HANDLER'='mygeohash', 
+'INDEX_HANDLER.mygeohash.type'='geohash',   
+'INDEX_HANDLER.mygeohash.sourcecolumns'='longitude, latitude',   
+'INDEX_HANDLER.mygeohash.originLatitude'='19.832277',   
+'INDEX_HANDLER.mygeohash.gridSize'='50',   
+'INDEX_HANDLER.mygeohash.minLongitude'='1.811865',   
+'INDEX_HANDLER.mygeohash.maxLongitude'='2.782233',   
+'INDEX_HANDLER.mygeohash.minLatitude'='19.832277',   
+'INDEX_HANDLER.mygeohash.maxLatitude'='20.225281',   
+'INDEX_HANDLER.mygeohash.conversionRatio'='1000000');
+```
```

Comment:
Please add one liner description for all the INDEX_HANDLER properties for better understanding.

Reply:
okay done add a chart